Spec/RealtimeClientConnection.swift

```diff
@@ -3028,26 +2989,165 @@ class RealtimeClientConnection: QuickSpec {
                     default:
                         fail("unhandled: \(fixtureMessage["expectedType"].string!)")
                     }
+                }
 
-                    waitUntil(timeout: testTimeout) { done in
-                        channel.publish([message]) { err in
-                            if let err = err {
-                                fail("\(err)")
+                // https://github.com/ably/wiki/issues/22
+                it("should encode and decode fixture messages as expected") {
+                    let options = AblyTests.commonAppSetup()
```

Comment:
The point here is that tests don't know how the internals of a library work and should not care.  The point of our interoperability tests was to check that a message when decoded/encoded via JSON or MsgPack are consistent with the types we expect to see. We agreed the only reliable way to get data into the Ably system was via a raw REST request to Ably, and then to check that it was as expected, was also via a raw REST history request to Ably and then to parse the JSON (or inspect the type if binary/string) and compare with what we see the library interprets that over JSON and MsgPack.  So if we want to test how our Ably library encoded/decodes objects with a JSON transport it should use that, and if we want to see how our Ably library does that MsgPack it should use a MsgPack transport.  This test uses MsgPack, so the JSON encoding / decoding of a message injected or received via a raw request is not being done. That is why in Ruby I have a set of tests [using the JSON transport](https://github.com/ably/ably-ruby/pull/94/files#diff-5a92463d6ab8ad3fc8cf8aff5cfde6f8R746) and then using the [MsgPack transport](https://github.com/ably/ably-ruby/pull/94/files#diff-5a92463d6ab8ad3fc8cf8aff5cfde6f8R798) but all the time using a [raw JSON request to inject or check the fixture in Ably](https://github.com/ably/ably-ruby/pull/94/files#diff-5a92463d6ab8ad3fc8cf8aff5cfde6f8R823) irrespective of the transport being tested for interoperability.